Cannot find `assert` in `core` compile error

<!--
If this is a security issue, please report it following the
[security reporting procedure](https://github.com/model-checking/kani/security/policy).
-->

<!--
Thank you for filing a bug report! 🐛
Please provide a short summary of the issue, along with the information necessary to replicate.
-->

Steps to reproduce:
1. `cargo new hdrhistogram_dep`
2. `cd hdrhistogram_dep`
3. `cargo add hdrhistogram`

using the following command line invocation:

```
cargo kani
```

with Kani version: 94b4e8459f8

I expected to see this happen: `No proof harnesses (functions with #[kani::proof]) were found to verify.`

<!--
If Kani crashed, please include a backtrace in the code block by setting
`RUST_BACKTRACE=1` in your environment. E.g. `RUST_BACKTRACE=1 kani ...`
-->
Instead, this happened:
```
$ cargo kani
   Compiling cfg-if v1.0.0
   Compiling memchr v2.5.0
   Compiling autocfg v1.1.0
   Compiling crc32fast v1.3.2
   Compiling crossbeam-utils v0.8.14
   Compiling adler v1.0.2
   Compiling minimal-lexical v0.2.1
   Compiling base64 v0.13.1
   Compiling byteorder v1.4.3
   Compiling miniz_oxide v0.6.2
   Compiling num-traits v0.2.15
   Compiling nom v7.1.1
   Compiling flate2 v1.0.25
   Compiling crossbeam-channel v0.5.6
   Compiling hdrhistogram v7.5.2
error[E0433]: failed to resolve: could not find `assert` in `core`
   --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/hdrhistogram-7.5.2/src/lib.rs:628:25
    |
628 |                         panic!("Tried to subtract value outside of range: {}", other_value);
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `assert` in `core`
    |
    = note: this error originates in the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `assert` in `core`
    --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/hdrhistogram-7.5.2/src/lib.rs:1649:9
     |
1649 | /         assert!(
1650 | |             high >= 2 * self.lowest_discernible_value,
1651 | |             "highest trackable value must be >= (2 * lowest discernible value)"
1652 | |         );
     | |_________^ could not find `assert` in `core`
     |
     = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0433`.
Error: "Failed to compile crate."
error: could not compile `hdrhistogram` due to 2 previous errors
Error: cargo exited with status exit status: 101
```

Here is a minimal reproducer:
```rust
mod core {
}

#[kani::proof]
fn main() {
    if kani::any() {
        let x = 5;
        panic!("x is {}", x);
    }
}
```

```
$ kani mod_core.rs 
error[E0433]: failed to resolve: could not find `assert` in `core`
 --> mod_core.rs:8:9
  |
8 |         panic!("x is {}", x);
  |         ^^^^^^^^^^^^^^^^^^^^ could not find `assert` in `core`
  |
  = note: this error originates in the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
Error: "Failed to compile crate."
Error: /home/ubuntu/git/kani/target/kani/bin/kani-compiler exited with status exit status: 1
```

--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -54,7 +54,7 @@ macro_rules! assert {
         // strategy, which is tracked in
         // https://github.com/model-checking/kani/issues/692
         if false {
-            core::assert!(true, $($arg)+);
+            ::core::assert!(true, $($arg)+);
         }
     }};
 }
@@ -158,7 +158,7 @@ macro_rules! unreachable {
     // handle.
     ($fmt:expr, $($arg:tt)*) => {{
         if false {
-            core::assert!(true, $fmt, $($arg)+);
+            ::core::assert!(true, $fmt, $($arg)+);
         }
         kani::panic(concat!("internal error: entered unreachable code: ",
         stringify!($fmt, $($arg)*)))}};
@@ -190,7 +190,7 @@ macro_rules! panic {
     // `panic!("Error: {}", code);`
     ($($arg:tt)+) => {{
         if false {
-            core::assert!(true, $($arg)+);
+            ::core::assert!(true, $($arg)+);
         }
         kani::panic(stringify!($($arg)+));
     }};

--- a/tests/kani/Panic/core_mod.rs
+++ b/tests/kani/Panic/core_mod.rs
@@ -1,0 +1,18 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This test checks that the panic macro does not cause a compiler error if
+//! there is a user-defined `core` module, which previously did (see
+//! https://github.com/model-checking/kani/issues/1984)
+
+mod core {}
+
+#[kani::proof]
+fn main() {
+    let x: u8 = kani::any();
+    let y = x / 2;
+    if y > x {
+        // impossible
+        panic!("y is {}", y);
+    }
+}

